### PR TITLE
Run2-hcx160 Allow Plan1 geometry to work for standard access of getPosition

### PR DESCRIPTION
**Pull request by bsunanda**
This is back port of PR #20868

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna Please approve this

File: Geometry/HcalTowerAlgo/interface/HcalGeometry.h [hunk 1]
```diff
@@ -100,19 +102,19 @@ class HcalGeometry : public CaloSubdetectorGeometry {
...
-			const CCGFloat*    parm,
-			const DetId&       detId     ) override ;
+		const GlobalPoint& f2 ,
+		const GlobalPoint& f3 ,
+		const CCGFloat*    parm,
+		const DetId&       detId     ) override ;
 
-  const CaloCellGeometry* getGeometry( const DetId& id ) const override {
-      return cellGeomPtr( m_topology.detId2denseId( id ) ) ;
+  const CaloCellGeometry* getGeometryBase( const DetId& id ) const {
```
Comment (ianna) [inline Geometry/HcalTowerAlgo/interface/HcalGeometry.h]: @bsunanda - I think, it should be private member function

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/HcalTowerAlgo/interface/HcalGeometry.h (hunk 1 above)
Comment (bsunanda) [reply Geometry/HcalTowerAlgo/interface/HcalGeometry.h]: I am using this in some testing code. That is why it is public. I am simply back porting here

Comment (bsunanda) [reply Geometry/HcalTowerAlgo/interface/HcalGeometry.h]: @iannna please look at the comment I made

Comment (ianna) [reply Geometry/HcalTowerAlgo/interface/HcalGeometry.h]: Let's keep API clean. Please, declare HcalGeometryPlan1Tester class as a friend if you plan to keep the test running.

**Comment by bsunanda** [comment]
I shall do that at the head. But these are simply back porting to older versions.
________________________________

File: Geometry/HcalTowerAlgo/interface/HcalGeometry.h (hunk 1 above)
Comment (ianna) [reply Geometry/HcalTowerAlgo/interface/HcalGeometry.h]: however, you are not back porting the test, so no need to have it public.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

**Pull request closed without merging**